torch/linalg/__init__.py

```diff
@@ -344,49 +381,78 @@
-          and MAGMA's `syevd` and `heevd` routines for CUDA inputs.
+.. note:: This function is computed using LAPACK's and MAGMA's `syevd` and `heevd` routines for
+          CPU and CUDA symmetric and Hermitian inputs respectively.
 
 .. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
 
-.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
-          a valid eigenvector. This function may compute different eigenvector representations on
-          different device types. Usually the difference is only in the sign of the eigenvector.
+.. warning:: The eigenvectors of a matrix are not unique. Any eigenvector may be multiplied by `-1`
+             in the real case or by :math:`e^{i \phi}` for any :math:`\phi \in \mathbb{R}` in the
+             complex case.
+             As such, the eigenvectors are not continuous functions of the inputs, and different
+             platforms, like NumPy, or inputs on different devices, may produce different
+             eigenvectors.
 
-.. note:: See :func:`torch.linalg.eigvalsh` for a related function that computes only eigenvalues.
-          However, that function is not differentiable.
+.. warning:: If `V` is used in the loss function, the gradient will only be finite when
+             :attr:`input` does not have repeated eigenvalues.
+             If the distance between any two eigenvalues is close to zero, the gradient with
+             respect to `V` will be numerically unstable, as it depends on
+             :math:`\frac{1}{\min_{i \neq j} \lambda_i - \lambda_j}`.
 
-Args:
-    input (Tensor): the Hermitian `n \times n` matrix or the batch of such matrices of size
-                    `(*, n, n)` where `*` is one or more batch dimensions.
-    UPLO ('L', 'U', optional): controls whether to use the upper-triangular or the lower-triangular part
-                               of :attr:`input` in the computations. Default is ``'L'``.
+.. seealso::
 
-Keyword args:
-    out (tuple, optional): tuple of two tensors to write the output to. Default is ``None``.
+        :func:`torch.linalg.eigvalsh` for a related function that computes only eigenvalues.
+        However, that function is not differentiable.
 
-Returns:
-    (Tensor, Tensor): A namedtuple (eigenvalues, eigenvectors) containing
+        :func:`torch.linalg.cholesky` for a different decomposition of a Hermitian matrix.
+        The Cholesky decomposition gives less information about the matrix but it is much faster
+        to compute than the eigenvalue decomposition.
+
+        :func:`torch.linalg.eig` for a (slower) function that computes the eigenvalue decomposition
+        of a not necessarily Hermitian matrix.
+
+        :func:`torch.linalg.svd` for a (slower) function that computes the more general SVD
+        decomposition of a not necessarily square matrix.
```

Comment:
"not necessarily square matrix" -> "of a matrix with any shape" ?

Reply:
I really like this wording! I've changed all the occurences to use this wording.